ingest-app/src/cmr/ingest/services/jobs_subscriptions.clj

```diff
@@ -0,0 +1,199 @@
+  (let [query-string-list (string/split query-string #"&")
+        query-map-list (map #(let [a (string/split % #"=")]
+                               {(first a) (second a)})
+                             query-string-list)]
+     (apply merge query-map-list)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Jobs for refreshing the collection granule aggregation cache in the indexer. This is a singleton job
+;; and the indexer does not have a database so it's triggered from Ingest and sent via message.
+;; Only one node needs to refresh the cache because we're using the  fallback cache with Redis cache.
+;; The value stored in Redis will be available to all the nodes.
+
+(defconfig email-server-host
+  "The host name for email server."
+  {:default ""
+   :type String})
+
+(defconfig email-server-port
+  "The port number for email server."
+  {:default 25
+   :type Long})
+
+(defconfig mail-sender
+  "The email sender's email address."
+  {:default ""
+   :type String})
+
+(defconfig email-subscription-processing-interval
+  "Number of seconds between jobs processing email subscriptions."
+  {:default 3600
+   :type Long})
+
+(defconfig email-subscription-processing-lookback
+  "Number of seconds to look back for granule changes."
+  {:default 3600
+   :type Long})
+
+(defn email-granule-url-list
+ "take a list of URLs and format them for an email"
+ [gran-ref-location]
+ (string/join "\n" (map #(str "* [" % "](" % ")") gran-ref-location)))
+
+(defn create-email-content
+ "Create an email body for subscriptions"
+ [from-email-address to-email-address gran-ref-location subscription]
+ (let [metadata (json/parse-string (:metadata subscription))
+       concept-id (get-in subscription [:extra-fields :collection-concept-id])
+       meta-query (get metadata "Query")
+       sub-start-time (:start-time subscription)]
+   {:from from-email-address
+    :to to-email-address
+    :subject "Email Subscription Notification"
+    :body [{:type "text/html"
+            :content (markdown/md-to-html-string
+                      (str "You have subscribed to receive notifications when data is added to the following query:\n\n"
+                           "`" concept-id "`\n\n"
+                           "`" meta-query "`\n\n"
+                           "Since this query was last run at "
+                           sub-start-time
+                           ", the following granules have been added or updated:\n\n"
+                           (email-granule-url-list gran-ref-location)
+                           "\n\nTo unsubscribe from these notifications, or if you have any questions, please contact us at [cmr-support@earthdata.nasa.gov](mailto:cmr-support@earthdata.nasa.gov).\n"))}]}))
+
+(defn- add-updated-since
+  "Pull out the start and end times from a time-constraint value and associate them to a map"
+  [raw time-constraint]
+  (let [parts (clojure.string/split time-constraint, #",")
```

Comment:
clojure.string is aliased as string, I think we should use it here. I know this is probably old but now would be a good time to fix it.